sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -194,6 +194,8 @@ void GetErrorAndThrow(const std::string& exceptionMessage)
 
 void WinHttpTransport::CreateSessionHandle(std::unique_ptr<_detail::HandleManager>& handleManager)
 {
+  handleManager->m_context.ThrowIfCancelled();
```

Comment:
yes, This one might not be needed:

The WinHttpOpen function is the first of the WinHTTP functions called by an application. It initializes internal WinHTTP data structures and prepares for future calls from the application. When the application finishes using the WinHTTP functions, it must call WinHttpCloseHandle to free the session handle and any associated resources.